docs/UserGuide.md

```diff
@@ -1,125 +1,151 @@
 
-Besides, define cache pool, cache allocation will happened in the pools.
-
-- shrink: whether shrink cache ways which already allocated to workload in
-          besteffort pool.
-- max_allowed_shared: max allowed workloads in shared cache pool
-- guarantee: allocate cache for workload max_cache == min_cache > 0
-- besteffort: allocate cache for workload max_cache > min_cache > 0
-- shared: allocate cache for workload max_cache == min_cache = 0
+* OSGroup: cache ways reserved for operating system usage.
+* InfraGroup: infrastructure tasks group, user can specify task binary name
+              the cache ways will be shared with other workloads
+* CachePool: cache allocation will happened in the pools.
+    - shrink: whether shrink cache ways which already allocated to workload in
+            "besteffort" pool.
+    - max_allowed_shared: max allowed workloads in shared cache pool
+    - guarantee: allocate cache for workload max_cache == min_cache > 0
+    - besteffort: allocate cache for workload max_cache > min_cache > 0
+    - shared: allocate cache for workload max_cache == min_cache = 0
 
 On a host which support max 20 cache ways, for this configuration file,
 we will have follow cache bit mask layout:
-
 ```
-
 OSGroup:    0000 0000 0000 0000 0001
 InfraGroup: 1111 1111 1111 1111 1110
 ```
+
 Available CBM in Cache Pools initially:
 ```
 guarantee:  0000 0000 0111 1111 1110
 besteffort: 0011 1111 1000 0000 0000
 shared:     1100 0000 0000 0000 0000
 ```
 
-## Create a workload
+### Prepare the credentials
 
-A workload could be a running task(s) or some cpus which want to allocate
-cache for them.
+For security considerations, RMD daemon runs the RESTful API service as a
+dedicated Unix user 'rmd' and performs privileged operations (PAM
+authentication or resctrl) as Unix 'root' user. RMD users need to prepare
+several things by themselves.
+
+1. Create user and group 'rmd' in the target Linux system
 
-The task(s)/cpus should be valided, RMD will fail your request they are
-invalid.
+    ```shell
+    $ sudo <TODO>
+    ```
+2. Ensure the following packages are installed on your target system for PAM
 
-Besides you need to specify what policy of the workload will be usage.
+    Debian or Ubuntu:
+    ```shell
+    $ sudo apt-get install openssl libpam0g-dev db-util
+    ```
+    Redhat(s)
+    ```shell
+    $ sudo dnf install openssl pam-devel db4-utils
+    ```
+3. Besides host Unix credentials, RMD can also use dedicated credentials setup
+   in a Berkeley database file.
 
-The post body could contains:
+    To run this script to setup users in Berkeley database:
+    ```shell
+    $ sudo ./scripts/setup_rmd_users.sh
+    ```
+    *Note: Only root user can setup or access users in Berkeley database*
 
+## Run the service
+
+Launch RMD manually, by specifying configuration directory:
+
+```shell
+$ sudo rmd --conf-dir /etc/rmd
 ```
+
+*RMD can be launched in debug mode that exposes RESTAPI service on HTTP.
```

Comment:
I think this is not correct, if user using default rmd.toml, it's not debug mode, the default setting in rmd.toml is tls enabled, see https://github.com/intel/rmd/blob/master/etc/rmd/rmd.toml#L15

By default , rmd will listen on 8443 with https protocol.

We can mention that if user wish to using debug mode ::

$ sudo rmd --conf-dir /etc/rmd --debug

And using --help to print all options, the options will overwrite de default 
$ rmd --help
Usage of rmd:
      --address string                   Listen address (default "localhost")
      --alsologtostderr                  log to standard error as well as files
      --clientauth string                The policy the server will follow for TLS Client Authentication (default "challenge")
      --conf-dir string                  Directy of config file
  -d, --debug                            Enable debug
      --debugport int                    Debug listen port (default 8081)
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --logtostderr                      log to standard error instead of files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
      --tlsport int                      TLS listen port (default 8443)
      --unixsock string                  Unix sock file path
  -v, --v Level                          log level for V logs
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging

There's some default value in rmd's code, which allow user to run rmd without /etc/rmd//rmd.toml. for now from my testing, the default value in the code is little different with the default rmd.toml, I will try to figure out and modify the default setting to make them behavior as same.